packages/assistant/src/RPA/Assistant/library.py

```diff
@@ -0,0 +1,991 @@
+
+        Adds an icon which can be used to indicate status
+        or the type of dialog being presented.
+
+        The currently supported icon types are:
+
+        ======= ==========================
+        Name    Description
+        ======= ==========================
+        Success A green check mark
+        Warning An orange warning triangle
+        Failure A red cross or X mark
+        ======= ==========================
+
+        The ``size`` of the icon can also be changed,
+        to a given height/width of pixels.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add icon              Warning    size=64
+            Add heading           Do you want to delete this order?
+            Add submit buttons    buttons=No,Yes
+            ${result}=    Run dialog
+        """
+        if not isinstance(variant, Icon):
+            variant = Icon(variant)
+
+        flet_icon_conversions: Dict[Icon, Tuple[str, str]] = {
+            Icon.Success: (icons.CHECK, colors.GREEN_500),
+            Icon.Warning: (icons.WARNING, colors.YELLOW_500),
+            Icon.Failure: (icons.CLOSE, colors.RED_500),
+        }
+        flet_icon, color = flet_icon_conversions[variant]
+
+        self._client.add_element(flet.Icon(name=flet_icon, color=color, size=size))
+
+    @keyword("Add text input", tags=["input"])
+    def add_text_input(
+        self,
+        name: str,
+        label: Optional[str] = None,
+        placeholder: Optional[str] = None,
+        rows: Optional[int] = None,
+    ) -> None:
+        """Add a text input element
+
+        :param name:        Name of result field
+        :param label:       Label for field
+        :param placeholder: Placeholder text in input field
+        :param rows:        Number of input rows
+
+        Adds a text field that can be filled by the user. The entered
+        content will be available in the ``name`` field of the result.
+
+        For customizing the look of the input, the ``label`` text can be given
+        to add a descriptive label and the ``placholder`` text can be given
+        to act as an example of the input value.
+
+        If the ``rows`` argument is given as a number, the input is converted
+        into a larger text area input with the given amount of rows by default.
+
+        Example:
+
+        .. code-block:: robotframework
+
+            Add heading    Send feedback
+            Add text input    email    label=E-mail address
+            Add text input    message
+            ...    label=Feedback
+            ...    placeholder=Enter feedback here
+            ...    rows=5
+            ${result}=    Run dialog
+            Send feedback message    ${result.email}  ${result.message}
+        """
+        # TODO: Do this in a cleaner way. Workaround because we use on_change
+        # handlers to record values, so default value otherwise will be missed
+        self._client.results[name] = placeholder
```

Comment:
Should there be some guarding (even if just warning logs) that multiple fields don't set the same result name?

Reply:
yes, we will add the guarding later on